gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+import logging
+import random
+import numpy
+
+import numpy as np
+from gensim.scripts.glove2word2vec import glove2word2vec
+from gensim.test.utils import get_tmpfile
+from gensim.models import KeyedVectors
+from collections import Counter
+from custom_losses import rank_hinge_loss
+from custom_layers import TopKLayer
+from sklearn.preprocessing import normalize
+from custom_callbacks import ValidationCallback
+from gensim import utils
+
+try:
+    import keras.backend as K
+    from keras import optimizers
+    from keras.losses import hinge
+    from keras.models import Model
+    from keras.layers import Input, Embedding, Dot, Dense, Reshape, Dropout
+    import tensorflow
+    tensorflow.set_random_seed(101010)
+    KERAS_AVAILABLE = True
+except ImportError:
+    KERAS_AVAILABLE = False
+
+random.seed(101010)
+numpy.random.seed(101010)
+logger = logging.getLogger(__name__)
+
+
+class DRMM_TKS(utils.SaveLoad):
+    """User friendly model for training on similarity learning data.
+    You only have to provide sentences in the data as a list of words.
+
+    Example Usage:
+    -------------
+    drmm_tks_model = DRMM_TKS_Model(queries, docs, labels, word_embedding_path)
+    drmm_tks_model.predict(test_queries, test_docs)
+
+    The data should have the format:
+    queries = ["When was World Wat 1 fought ?".split(),
+             "When was Gandhi born ?".split()]
+
+    docs = [
+            ["The world war was bad".split(),
+            "It was fought in 1996".split()],
+            ["Gandhi was born in the 18th century".split(),
+             "He fought for the Indian freedom movement".split(),
+             "Gandhi was assasinated".split()]
+           ]
+
+    labels = [[0, 1],
+              [1, 0, 0]]
+    """
+
+    def __init__(self, queries, docs, labels, word_embedding_path=None,
+                 text_maxlen=200, keep_full_embedding=True, normalize_embeddings=True,
+                 epochs=10, unk_handle_method='zero', validation_data=None):
+        """Initializes the model and trains it
+
+        Parameters:
+        -----------
+        queries: list of list of string words
```

Comment:
`iterable of list of str` (here and everywhere, we talked about online learning)

Reply:
also, should be optional (if I want to train it later myself)